elements/lisk-transactions/src/transfer.ts

```diff
@@ -140,7 +151,17 @@ export const transfer = (inputs: TransferInputs): Partial<TransactionJSON> => {
 		transactionWithSenderInfo,
 	);
 
-	transferTransaction.sign(passphrase);
+	if (passphrase) {
+		transferTransaction.sign(networkIdentifier, passphrase);
+
+		return transferTransaction.toJSON();
+	}
+
+	if (passphrases && keys) {
+		transferTransaction.sign(networkIdentifier, undefined, passphrases, keys);
+
+		return transferTransaction.toJSON();
+	}
 
-	return transferTransaction.toJSON();
+	return transaction;
```

Comment:
why we don't return `transactionWithSenderInfo`?

Reply:
Changed to return that